.github/workflows/release.yml

```diff
@@ -20,9 +20,64 @@ on:
         required: false
         type: string
   push:
-    tags: ["v*"]
+    tags: ['v*']
+  pull_request:
+    types: [closed]
+    branches: [main]
 
 jobs:
+  validate-release-pr:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true
+    outputs:
+      has_release_label: ${{ steps.check_label.outputs.has_release_label }}
+    steps:
+      - name: Check if PR has release label
+        id: check_label
+        run: |
+          if [[ ${{ contains(github.event.pull_request.labels.*.name, 'release') }} ]]; then
+            echo "This workflow was triggered by the release PR merge"
+            echo "has_release_label=true" >> $GITHUB_OUTPUT
+          else
+            echo "PR does not have 'release' label. This workflow should only run for release PRs."
+            echo "has_release_label=false" >> $GITHUB_OUTPUT
+          fi
+
+  create-github-release:
+    runs-on: ubuntu-latest
+    needs: validate-release-pr
+    if: github.event.pull_request.merged == true && needs.validate-release-pr.outputs.has_release_label == 'true'
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
should we check out the exact commit that was merged, and not the commit to main? 

My concern is timing and if somehow multiple commits get merged before the action runs.